pandas/core/generic.py

```diff
@@ -6068,14 +6068,14 @@ def fillna(
                 )
 
             elif isinstance(value, (dict, ABCSeries)):
+                result = self if inplace else self.copy()
+
                 if axis == 1:
-                    raise NotImplementedError(
-                        "Currently only can fill "
-                        "with dict/Series column "
-                        "by column"
-                    )
+                    for label in self.columns:
```

Comment:
Accessing by label like typically causes issues when users have duplicate label names - have you tested that case by chance?

Reply:
@WillAyd 
I didn't. Thanks for pointing out. i change the logic and add test cases.